web/src/main/java/se/inera/intyg/webcert/web/service/unansweredcommunication/UnansweredCommunicationServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+import se.inera.intyg.clinicalprocess.healthcond.certificate.getcertificateadditions.v1.StatusType;
+import se.inera.intyg.webcert.persistence.arende.model.Arende;
+import se.inera.intyg.webcert.persistence.arende.model.ArendeAmne;
+import se.inera.intyg.webcert.persistence.model.Status;
+import se.inera.intyg.webcert.web.service.arende.ArendeService;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredCommunicationRequest;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredCommunicationResponse;
+import se.inera.intyg.webcert.web.web.controller.internalapi.dto.UnansweredQAs;
+
+@Service
+public class UnansweredCommunicationServiceImpl implements UnansweredCommunicationService {
+
+    private final ArendeService arendeService;
+    private static final String FK = "FK";
+
+    public UnansweredCommunicationServiceImpl(ArendeService arendeService) {
+        this.arendeService = arendeService;
+    }
+
+    @Override
+    public UnansweredCommunicationResponse get(UnansweredCommunicationRequest request) {
+        final var maxDaysOfUnansweredCommunication = request.getMaxDaysOfUnansweredCommunication();
+        final var patientIds = request.getPatientIds();
+
+        if (patientIds == null || patientIds.isEmpty() || maxDaysOfUnansweredCommunication == null) {
+            throw new IllegalArgumentException(String.format(
+                "Request is missing parameters required to make request. patientIds '%s' maxDaysOfUnansweredCommunication '%s'",
+                patientIds,
+                maxDaysOfUnansweredCommunication));
+        }
+
+        final var arendenForPatients = arendeService.getArendenForPatientsWithTimestampAfterDate(
+            formatPatientIds(patientIds),
+            LocalDateTime.now().minusDays(maxDaysOfUnansweredCommunication)
+        );
+
+        if (arendenForPatients.isEmpty()) {
+            return new UnansweredCommunicationResponse(Collections.emptyMap());
+        }
+
+        return new UnansweredCommunicationResponse(createUnansweredCommunication(arendenForPatients));
+    }
+
+    private static List<String> formatPatientIds(List<String> patientIds) {
+        return patientIds.stream()
+            .map(patientId -> patientId.replace("-", ""))
+            .collect(Collectors.toList());
+    }
+
+    private Map<String, UnansweredQAs> createUnansweredCommunication(List<Arende> arenden) {
+        final var unansweredCommunication = new HashMap<String, UnansweredQAs>();
+        arenden.forEach(arende -> getUnansweredCommunication(arende, unansweredCommunication));
+        return unansweredCommunication;
+    }
+
+    private void getUnansweredCommunication(Arende arende, Map<String, UnansweredQAs> unansweredCommunication) {
+        final var status = convertStatus(arende);
+        if (isNotUnansweredCommunication(arende, status)) {
+            return;
+        }
+        final var unansweredQAs = getUnansweredQA(arende, unansweredCommunication);
```

Comment:
See comment on getUnansweredQA. Maybe easier to just to remove the additional private method, when you can call the map directly.